proposals/xxxx-package-manager-manifest-api-redesign.md

```diff
@@ -0,0 +1,377 @@
+
+* Proposal: [SE-XXXX](xxxx-package-manager-manifest-api-redesign.md)
+* Author: [Ankit Aggarwal](https://github.com/aciidb0mb3r)
+* Review Manager: TBD
+* Status: **Discussion**
+
+## Introduction
+
+This proposal is a redesign of `Package.swift` manifest APIs provided by Swift Package Manager.  
+Note: It does not aim to change any actual underlying functionality.
+
+## Motivation
+
+`Package.swift` manifest API was designed prior to the [API Design Guidelines]
+(https://swift.org/documentation/api-design-guidelines/), and its design was not reviewed by the
+evolution process. We would like to redesign this API as necessary to provide a clean,
+conventions-compliant API that we can rely on in the future.
+
+## Proposed solution
+
+This is how a regular manifest looks like currently:
+
+```swift
+let package = Package(
+    name: "Paper",
+    targets: [
+        // Helper tool.
+        Target(
+            name: "tool",
+            dependencies: [
+                "Paper",
+                "SwiftyJSON"
+            ]),
+        // The main library.
+        Target(
+            name: "Paper",
+            dependencies: [
+                .ByName(name: "Basic"),
+                .Target(name: "Utility"),
+                .Product(name: "CHTTPParser", package: nil),
+            ])
+    ],
+    products: [
+        .Executable(name: "tool", targets: ["tool"]),
+        .Libary(name: "Paper", type: .static, targets: ["Paper"]),
+        .Libary(name: "PaperDy", type: .dynamic, targets: ["Paper"]),
+    ],
+    dependencies: [
+        .Package(url: "http://github.com/SwiftyJSON", majorVersion: 1),
+        .Package(url: "../CHTTPParser", majorVersion: 1),
+        .Package(url: "http://some/other/lib", Version(1, 2, 3)),
+    ]
+    exclude: [
+        "Tests/Fixtures",
+    ]
+)
+```
+
+A system package manifest:
+
+```swift
+let package = Package(
+    name: "Clibgit",
+    pkgConfig: "libgit2",
+    providers: [
+        .Brew("libgit2"),
+        .Apt("libgit2-dev"),
+    ]
+)
+```
+
+---
+
+### Case conventions:
+
+> Names of types and protocols are UpperCamelCase. Everything else is lowerCamelCase.
+
+APIs incorrectly using upper camel cases should be fixed:
+
+* `Product`:
+
+```swift
+enum Product {
+    static func executable(name: String, targets: [String]) -> Product
+    static func library(name: String, type: LibraryProduct.LibraryType? = nil, targets: [String]) -> Product
+}
+```
+
+* `Target.Dependency`:
+
+```swift
+class Target {
+    enum Dependency {
+        case target(name: String)
+        case product(name: String, package: String?)
+        case byName(name: String)
+    }
+}
+```
+
+* `Package.Dependency`:
+
+```swift
+final class Package {
+    class Dependency {
+        static func package(...) -> Dependency  // Represents all static methods having name `Package`.
+    }
+}
+```
+
+### `Product`:
+
+// I propose we turn the tagets parameter on the static methods variadic, so brackets can be ommited.  
```

Comment:
s/tagets/targets/

Are the comments comments '//' things you aren't sure about?